### ISSUE TITLE
optimize migration tool

### ISSUE THREAD

**Issue by cookie33** [EUDAT-B2SAFE/B2HANDLE-Migration#3]
Hi,
For the handle: `842/000017b2-02ab-11e6-84a7-fa163e62896a`
```
curl http://epic3.storage.surfsara.nl:8001/api/handles/842/000017b2-02ab-11e6-84a7-fa163e62896a?pretty
{
  "responseCode": 1,
  "handle": "842/000017b2-02ab-11e6-84a7-fa163e62896a",
  "values": [
    {
      "index": 1,
      "type": "URL",
      "data": {
        "format": "string",
        "value": "irods://130.186.13.14:1247/cinecaDMPZone/home/irods#devRZG/nested00/testn00_1k4057.dat"
      },
      "ttl": 86400,
      "timestamp": "2016-04-15T01:41:00Z"
    },
    {
      "index": 2,
      "type": "10320/LOC",
      "data": {
        "format": "string",
        "value": "<locations><location href=\"irods://130.186.13.14:1247/cinecaDMPZone/home/irods#devRZG/nested00/testn00_1k4057.dat\" id=\"0\"/></locations>"
      },
      "ttl": 86400,
      "timestamp": "2016-04-15T01:41:00Z"
    },
    {
      "index": 3,
      "type": "CHECKSUM",
      "data": {
        "format": "string",
        "value": "e97db5668d9413b7a250a005a6fa7ff5"
      },
      "ttl": 86400,
      "timestamp": "2016-04-15T01:41:00Z"
    },
    {
      "index": 4,
      "type": "EUDAT/ROR",
      "data": {
        "format": "string",
        "value": "843/3c6186a2-a408-11e5-97b6-0050569ed64e"
      },
      "ttl": 86400,
      "timestamp": "2016-04-15T01:41:00Z"
    },
    {
      "index": 5,
      "type": "EUDAT/PPID",
      "data": {
        "format": "string",
        "value": "843/3c6186a2-a408-11e5-97b6-0050569ed64e"
      },
      "ttl": 86400,
      "timestamp": "2016-04-15T01:41:00Z"
    },
    {
      "index": 100,
      "type": "HS_ADMIN",
      "data": {
        "format": "admin",
        "value": {
          "handle": "0.NA/842",
          "index": 200,
          "permissions": "011111110011"
        }
      },
      "ttl": 86400,
      "timestamp": "2016-04-15T01:41:00Z"
    }
  ]
}
```
At the moment the script does:
```
REMOVE 4:842/000017b2-02ab-11e6-84a7-fa163e62896a
REMOVE 5:842/000017b2-02ab-11e6-84a7-fa163e62896a
REMOVE 2:842/000017b2-02ab-11e6-84a7-fa163e62896a
MODIFY 842/000017b2-02ab-11e6-84a7-fa163e62896a
3 EUDAT/CHECKSUM 86400 1110 UTF8 e97db5668d9413b7a250a005a6fa7ff5

ADD 842/000017b2-02ab-11e6-84a7-fa163e62896a
1110 EUDAT/CHECKSUM_TIMESTAMP 86400 1110 UTF8 2016-04-15T03:41:00
1010 EUDAT/FIXED_CONTENT 86400 1110 UTF8 FALSE
1120 EUDAT/ROR 86400 1110 UTF8 843/3c6186a2-a408-11e5-97b6-0050569ed64e
1130 EUDAT/FIO 86400 1110 UTF8 843/3c6186a2-a408-11e5-97b6-0050569ed64e
1150 EUDAT/PARENT 86400 1110 UTF8 843/3c6186a2-a408-11e5-97b6-0050569ed64e
1000 EUDAT/PROFILE_VERSION 86400 1110 UTF8 1
```
It deletes
* 2  10320/LOC
* 4  EUDAT/ROR
* 5  EUDAT/PPID
And it adds them later with different indexes.

It seems more economical to do following:
leave `EUDAT/ROR` or if it is only `ROR` update it to `EUDAT/ROR`
update EUDAT/PPID to EUDAT/PARENT
```
MODIFY 842/000017b2-02ab-11e6-84a7-fa163e62896a
5 EUDAT/PARENT 86400 1110 UTF8 843/3c6186a2-a408-11e5-97b6-0050569ed64e
```

**Comment by TobiasWeigel** [comment]
The reason for the add/remove instead of modify statements is to set the same index values for the same fields across all Handles. I suggest to wait with changes until we know whether the batch duration is a problem.

**Comment by cookie33** [comment]
But the new handles generated with the handle library will get an key/value with  index starting from 1.

**Comment by TobiasWeigel** [comment]
Fixed in ce5f8dc534d1c33a266c9c310eba60975ecfd8bb
